src/msg.c

```diff
@@ -524,6 +524,14 @@ static int8_t msg_auth(struct session *session,
 
 	memcpy(uuid, kmauth->uuid, sizeof(kmauth->uuid));
 	memcpy(token, kmauth->token, sizeof(kmauth->token));
+
+	/* Verify if has uuid and token before assign to them */
+	if (session->uuid)
+		l_free(session->uuid);
+
+	if (session->token)
+		l_free(session->token);
```

Comment:
I think that this code is unnecessary. Because session->uuid/token always will be NULL.
But the if conditional needs to be placed on session_destroy function.

Reply:
You are right. It's not necessary